signac/contrib/job.py

```diff
@@ -359,11 +355,36 @@ def _statepoint_filename(self):
         # use str.join with os.sep instead of os.path.join for speed.
         return os.sep.join((self.workspace(), self.FN_MANIFEST))
 
-    @property
+    # Decorated properties aren't supported: https://github.com/python/mypy/issues/1362
+    @property  # type: ignore
```

Comment:
Can this be decorated in the opposite order? Aside from the mypy issue, it feels like putting `@property` closer to `def ws(self):` would be clearer.

Reply:
Unfortunately no, I tried this before and it doesn't work. `deprecation.deprecated` only works on a callable.